Library/Homebrew/cmd/test-bot.rb

```diff
@@ -584,7 +591,9 @@ def formula(formula_name)
             test "brew", "install", bottle_filename
           end
         end
-        test "brew", "test", "--verbose", formula_name if formula.test_defined?
+        shared_test_args = ["--verbose"]
+        shared_test_args << "--keep-tmp" if ARGV.keep_tmp?
+        test "brew", "test", *shared_test_args, formula_name if formula.test_defined?
```

Comment:
Unfortunately, this doesn't work with Ruby 1.8. The array splat (`*`) only works when used with the last argument. (Same for other instances of this pattern further down.)